nemo/collections/tts/models/fastpitch.py

```diff
@@ -329,24 +341,43 @@ def generate_spectrogram(
         return spect
 
     def training_step(self, batch, batch_idx):
-        attn_prior, durs, speaker = None, None, None
+        attn_prior, durs, speaker, energy = None, None, None, None
         if self.learn_alignment:
             if self.ds_class_name == "TTSDataset":
-                if SpeakerID in self._train_dl.dataset.sup_data_types_set:
-                    audio, audio_lens, text, text_lens, attn_prior, pitch, _, speaker = batch
-                else:
-                    audio, audio_lens, text, text_lens, attn_prior, pitch, _ = batch
+                batch_dict = process_batch(batch, self._train_dl.dataset.sup_data_types_set)
+                audio = batch_dict.get("audio")
+                audio_lens = batch_dict.get("audio_lens")
+                text = batch_dict.get("text")
+                text_lens = batch_dict.get("text_lens")
+                attn_prior = batch_dict.get("align_prior_matrix", None)
+                pitch = batch_dict.get("pitch", None)
+                energy = batch_dict.get("energy", None)
+                speaker = batch_dict.get("speaker_id", None)
             else:
                 raise ValueError(f"Unknown vocab class: {self.vocab.__class__.__name__}")
         else:
             audio, audio_lens, text, text_lens, durs, pitch, speaker = batch
```

Comment:
I wonder if we can make the logic more concise like below, @rlangman could you pls confirm?

``` python
        ...
        batch_dict = process_batch(batch, self._train_dl.dataset.sup_data_types_set)
        audio = batch_dict.get("audio")
        audio_lens = batch_dict.get("audio_lens")
        text = batch_dict.get("text")
        text_lens = batch_dict.get("text_lens")
        attn_prior = batch_dict.get("align_prior_matrix", None)
        pitch = batch_dict.get("pitch", None)
        energy = batch_dict.get("energy", None)
        speaker = batch_dict.get("speaker_id", None)
        dur = batch_dict.get("dur", None)

        if self.learn_alignment:
            assert self.ds_class_name == "TTSDataset", f"Unknown vocab class: {self.vocab.__class__.__name__}"
```

Reply:
Done.

Reply:
Done.